### ISSUE TITLE
Customise a join study url

### ISSUE THREAD

**Issue by asiripanich** [e-mission/e-mission-docs#508]
Could you give us a direction how to test if extra arguments are working in the devapp?

For example, we would like to reward=reward1 as the following:

emission://change_client?new_client=rkunsw&clear_local_storage=true&clear_usercache=true&reward=1

**Comment by shankari** [comment]
**Short answer**: the current way to do this is to use separate channels for the different rewards
**Long answer**: the code which deals with parsing the parameters that are passed in from the channel is https://github.com/e-mission/e-mission-phone/blob/master/www/js/splash/updatecheck.js#L92
however, that code is run in emTripLog, so handling additional parameters would require updating emTripLog, which I would recommend against just before you launch your study
**Long-term enhancement**: This request actually brings up an enhancement for the next emTripLog update, which will hopefully happen after your study. The parameter handling code [currently stores the `new_client` parameter so that it is available to the channel-specific code](https://github.com/e-mission/e-mission-phone/blob/master/www/js/splash/updatecheck.js#L98) even after the UI is reloaded. However, it doesn't store any of the other parameters passed in - mainly because we haven't had a use case for that yet. It should be pretty easy to, in the next update of emTripLog, ensure that all the parameters are stored for use later.

**Comment by shankari** [comment]
The tripaware study performed a randomized control test (RCT) comparing emotion, information and a control group. They randomly assigned users to different groups when they visited the study page and customized the URL accordingly. Their customization code is here as a template:
https://github.com/e-mission/e-mission-server/blob/urap-2017/webapp/www/js/new-age.js#L45

and here's what it looks like in practice. Refresh the page a couple of times to see the UI change...
https://tripaware.damajash.org/

**Comment by shankari** [comment]
> the current way to do this is to use separate channels for the different rewards

I am happy to create multiple branches and channels for you so we can get this to work in the short term. LMK what names you want.

**Comment by asiripanich** [comment]
Thanks @shankari for your quick response. We need four branches - rciti1, rciti2, rciti3, rciti4.

**Comment by shankari** [comment]
ok, I just created 4 branches:
https://github.com/e-mission/e-mission-phone/tree/rciti1
https://github.com/e-mission/e-mission-phone/tree/rciti2
https://github.com/e-mission/e-mission-phone/tree/rciti3
https://github.com/e-mission/e-mission-phone/tree/rciti4

**Comment by asiripanich** [comment]
Does this mean that I need to deploy 4 instances of emission server, one for each branch?

**Comment by shankari** [comment]
No, all the clients can be configured to point to the same server. That's what the tripaware team did.

**Comment by asiripanich** [comment]
Thanks Shankari! 

@atton16 could you have a look into this?
